libp2p/transports/tortransport.nim

```diff
@@ -272,6 +272,35 @@ proc new*(
     torSwitch.connManager.peerStore = switch.peerStore
     return torSwitch
 
+proc new*(
+  T: typedesc[TorSwitch],
+  torServer: TransportAddress,
+  rng: ref HmacDrbgContext,
+  addresses: seq[MultiAddress] = @[],
+  flags: set[ServerFlags] = {},
+  seckey: PrivateKey ): TorSwitch
```

Comment:
Could you explain why you need to specify a PrivateKey for your usecase?
Also, I don't think having a new constructor for this is necessary, you could just add a
`seckey = none(PrivateKey)`
To the first one, and something like
```nim
seckey.withValue(seckey):
  build.withPrivateKey(seckey)
```